### PR TITLE
Introduced Stack() and Stack.exports 

### PR DESCRIPTION
**Pull request by lbialy**
This replaces faulty final for-comprehension (broken previews). Updated integration tests, docs, examples and templates.

### CODE REVIEW

**Comment by lbialy** [comment]
closes #340

File: core/src/main/scala/besom/internal/Stack.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package besom.internal
+
+import com.google.protobuf.struct.Struct
+
+case class Stack private[besom] (private val _exports: Exports, private val dependsOn: Vector[Output[?]]):
```
Comment (pawelprazak) [inline core/src/main/scala/besom/internal/Stack.scala:5]: needs scaladoc, please

File: core/src/main/scala/besom/internal/Stack.scala [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  private[besom] def evaluateDependencies(using Context): Result[Unit] =
+    Output.sequence(dependsOn).getData.void
+
+  private[besom] def getExports: Exports = _exports
+
+  private[besom] def getDependsOn: Vector[Output[?]] = dependsOn
+
+  def exports: Export = Export(this)
+
+object Stack:
```
Comment (pawelprazak) [inline core/src/main/scala/besom/internal/Stack.scala:15]: needs scaladoc, please

File: core/src/main/scala/besom/internal/StackFactory.scala [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package besom.internal
+
+trait StackFactory:
+  val exports: EmptyExport.type = EmptyExport
+
+  def apply(dependsOn: Output[?]*)(using Context): Stack =
```
Comment (pawelprazak) [inline core/src/main/scala/besom/internal/StackFactory.scala:6]: needs scaladoc, please

**Review by pawelprazak — APPROVED** [review]
Looks great, thank you, just a couple of nits :)

File: core/src/main/scala/besom/internal/Stack.scala (hunk 2 above)
Comment (lbialy) [reply core/src/main/scala/besom/internal/Stack.scala:15]: this one is completely internal

File: core/src/main/scala/besom/internal/StackFactory.scala (hunk 3 above)
Comment (lbialy) [reply core/src/main/scala/besom/internal/StackFactory.scala:18]: added

File: core/src/main/scala/besom/internal/Stack.scala (hunk 1 above)
Comment (lbialy) [reply core/src/main/scala/besom/internal/Stack.scala:14]: added

File: core/src/main/scala/besom/internal/StackFactory.scala (hunk 3 above)
Comment (pawelprazak) [reply core/src/main/scala/besom/internal/StackFactory.scala:18]: thank you!

**Review by pawelprazak — APPROVED** [review]

**Pull request merged**